### ISSUE TITLE
XSS vulnerabilities in Codiad-2.8.4

### ISSUE THREAD

**Issue by chluo1997** [Codiad/Codiad#1132]
Hi, I found multiple XSS vulnerabilities in Codiad-2.8.4.

Detail:
1. path: Codiad-2.8.4/components/user/dialog.php
    parameter: ?action=projects&username=<script>alert(1)</script>
<img width="910" alt="xss2" src="https://user-images.githubusercontent.com/70422882/130075489-2ecf8e41-d0c7-45bc-919a-e8d22d49532d.png">

2. path: Codiad-2.8.4/components/active/dialog.php
    parameter: action=confirm&path=<script>alert(1)</script>
<img width="969" alt="xss1" src="https://user-images.githubusercontent.com/70422882/130075534-f37c50db-16fb-4119-97ab-aa2f10bb8048.png">

**Comment by HLSiira** [comment]
You should check the last two opened issues, they go into major detail about the above XSS vuls in Codiad. Almost every component is vulnerable to it, and most importantly the file manager. IIRC You can name a file to install `evil script` and then an admin account who looks at the folder will cause all sorts of bad stuff.

Nice catch though, I didn't notice it much when I forked Codiad

**Comment by chluo1997** [comment]
#1122 reported XSS vulnerability in file controller.php and I report XSS vulnerabilities in two dialog.php. I  think they are different things.
I cannot see the contents in #1131 and have no idea of their discovery. Here I post two PoCs to demonstrate the XSS vuls confirmed by me.

**Comment by HLSiira** [comment]
Yes, if that's what you want to do, but you may as well start opening XSS issues for every single file in Codiad. The method by which Codiad handles sending data is inherently vulnerable, which is what #1122 is demonstrating.

**Comment by seongil-wi** [comment]
Hello! I already reported this issue via email (dev@codiad.com and ksafranski@gmail.com) in April and got a CVE (CVE-2021-30217).
